tools/cldr-code/src/main/java/org/unicode/cldr/util/NameType.java

```diff
@@ -0,0 +1,275 @@
+package org.unicode.cldr.util;
+
+import java.util.Locale;
+
+public enum NameType {
+    NONE,
+    LANGUAGE,
```

Comment:
These are the replacements for CLDRFile.NO_NAME, CLDRFile.LANGUAGE_NAME, etc. (which were integers, NOT enum)

Reply:
I really like putting this in an enum :D

However, I'm having a hard time understanding what NameType means -- can you add a longer comment about what it means? I wonder if there's a better enum name.

Reply:
As I just wrote in another reply, "A path for a language name is associated with NameType.LANGUAGE; a path for a script name is associated with NameType.SCRIPT; etc."

There are different types of names, and this enum enumerates those types, so I think it's a pretty good name... You're right, though, adding a comment here could help

Reply:
could be a future PR to add some brief enum docs (at the enum level)